lib/gringotts/gateways/pin_pay.ex

```diff
@@ -0,0 +1,423 @@
+defmodule Gringotts.Gateways.Pinpay do
+  @moduledoc """
+  [PinPay][home] gateway implementation.
+
+  The login credentials are:
+
+  | Key      | Credentials        |
+  | ------   | --------           |
+  | username | `api_key`          |
+  | password | ``                 |
+
+
+  The following features of PinPayments are implemented:
+
+  | Action                       | Method        |
+  | ------                       | ------        |
+  | Authorize                    | `authorize/3` |
+  | Capture                      | `capture/3`   |
+  | Purchase                     | `purchase/3`  |
+  | Store                        | `store/2`     |
+  | Refund                       | `refund/3`    |
+  | Respond                      | `respond/1`   |
+
+
+   ## The `opts` argument
+
+  Most `Gringotts` API calls accept an optional `keyword` list `opts` to supply
+  optional arguments for transactions with the PINPAY gateway. The following keys
+  are supported:
+
+  | Key               | Type       | Remark                                           |
+  | ----              | ----       | ---                                              |
+  | `address`         | `map`      | The address of the customer                      |
+  | `email_id`        | `String.t` | Merchant provided email addres                   |
+  | `description`     | `String.t` | Merchant provided description of the transaction |
+  | `ip_address`      | `String.t` | Merchant provided ip address (optional)                     |
+
+
+  > PINPAY supports more optional keys and you can raise an [issue][issues] if
+    this is important to you.
+
+  [issues]: https://github.com/aviabird/gringotts/issues/new
+
+
+  ### Schema
+
+  * `address` is a `map` from `atoms` to `String.t`, and can include any
+    of the keys from:
+    `[:street1, :street2, :city, :region, :postal_code, :country, :phone, ]`
+
+
+  ## Registering your PINPAY account at `Gringotts`
+
+  | Config parameter | PINPAY secret       |
+  | -------          | ----              |
+  | `:username`      | **API_SECRET_KEY**  |
+  | `:password`      | Empty string        |
```

Comment:
Empty String?

Reply:
yes, they have mentioned password to be an empty string.

Reply:
Then this should not be a config parameter, as we know that this is always empty.